src/TRoom.cpp

```diff
@@ -245,15 +247,12 @@ void TRoom::setExitWeight(const QString& cmd, int w)
 {
     if (w > 0) {
         exitWeights[cmd] = w;
-        if (mpRoomDB && mpRoomDB->mpMap) {
-            mpRoomDB->mpMap->mMapGraphNeedsUpdate = true;
-        }
+        mpRoomDB->mpMap->mMapGraphNeedsUpdate = true;
```

Comment:
@SlySven wasn't the `if` check here redundant?

Reply:
IIRC At one point in the past *I think*, during XML map imports `TRoom`s were created without being associated with a `TRoomDB` until all the details are filled out - so there were corner-cases where `mpRoom` could be a `nullptr` - reviewing all the cases where there are `new TRoom(....)` now that is not the case.

tl;dr; Yeah it is redundant now.